src/server/handlers.js

```diff
@@ -45,22 +45,77 @@ module.exports = {
   submit: function(req, res, next) {
     module.exports.getRoutes(req.body.start, req.body.end)
     .then(results => {
-      res.send(results);
+      // Parse nested object returned by Google's API to
+      // specifically get Array of routes.
+      var routesArray = JSON.parse(results.body).routes;
+
+      // Call getRestaurants along the returned route.
+      module.exports.getRestaurants(req, res, routesArray);
     })
     .catch(err => {
       console.log('Error requesting routes: ', err);
       res.end();
-    })
+    });
   },
 
   /*
-   * Input: Object
+   * Input: Array
    * Output: Promise
-   * Description: Takes a parameters object and returns a list
-   *              of yelp provided data.
+   * Description: Takes in the route object returned by Google's API,
+   *              and returns an array of restaurant objects from Yelp.
    */
-  getRestaurants: (customParams) => {
-    return yelp.search(customParams);
+  getRestaurants: (req, res, routesArray) => {
+
+    // Object to be returned to the client. 
+    // Stores route and restaurants in two seperate arrays.
+    var responseObject = {
+      route: routesArray,
+      restaurants: [],
+    };
+
+    // Grab the array of steps out of Googles nested objects.
+    var stepsArray = routesArray[0].legs[0].steps;
+
+    // Keeps track of the number of Yelp queries we've made.
+    var queryCounter = 0;
+    
+
+    // Makes a unique Yelp query for each step along the given route.
+    stepsArray.forEach(function (step, index) {
+
+      // Calculate the geographical midpoint along each step of the journey.
+      let midpointLatitude = (step.start_location.lat + step.end_location.lat) / 2;
+      let midpointLongitude = (step.start_location.lng + step.end_location.lng) / 2;
+
+      // Establish parameters for each individual yelp query.
+      let searchParameters = {
+        'radius_filter': step.distance.value / 2,
+        'll': `${midpointLatitude},${midpointLongitude}`,
+        'term': 'food',
+      };
+
+      // Query Yelp's API.
+      yelp.search(searchParameters)
+
+        // Sucess callback
+        .then(function (searchResults) {
+          // Add the returned businessees to the restauraunts array.
+          responseObject.restaurants = responseObject.restaurants.concat(searchResults.businesses);
+          
+          // Send a response to the client if all requisite queries have been made.
+          queryCounter++;
+          queryCounter >= stepsArray.length ? res.send(responseObject) : null;
+        }) 
+
+        // Error callback
+        .catch(function (error) {
+          console.log('Yelp returned an error:', error);
+
+          // Send a response to the client if all requisite queries have been made.
```

Comment:
We probably want to clear out the routes array and restaurants for each server request, so the client wouldn't receive all previous results